build.gradle

```diff
@@ -103,9 +113,29 @@ subprojects {
     version = projectVersion
 
     if ( project.javaToolchainEnabled ) {
-        compileJava.options.release = project.baselineJavaVersion.asInt()
-        compileTestJava.options.release = project.baselineJavaVersion.asInt()
-        compileJava.options.encoding = 'UTF-8'
+        // "sourceCompatibility" is not supported with toolchains. We have to work around that limitation.
+        compileJava.configure {
+            if (project.compileJdkVersion.asInt() < 11) {
```

Comment:
Wasn't `release` introduced in 9?

Reply:
It was. I added a commit. We don't test Java 9 / 10 though: they are not supported anywhere AFAIK.

Reply:
:)
Thanks, I was just checking, I didn't mean that you should fix it.